ui/app/common/pnc-client/message-bus/messageBus.js

```diff
@@ -50,6 +50,20 @@
       messageBus.onGroupBuildProgressChange(newGroupBuildProgressEventHandler(events.GROUP_BUILD_PROGRESS_CHANGED));
       messageBus.onGroupBuildStatusChange(newGroupBuildProgressEventHandler(events.GROUP_BUILD_STATUS_CHANGED));
 
+      function newGenericSettingProgressEventHandler(eventType) {
+        $rootScope.$broadcast(eventType);
+      }
+
+      messageBus.onGenericSettingMaintenanceChanged(notification => {
+        if (notification.message) {
+          let message = JSON.parse(notification.message);
+          if (message.maintenanceModeEnabled) {
+            newGenericSettingProgressEventHandler(events.MAINTENANCE_MODE_ON);
```

Comment:
I think here you can just invoke $rootScope.$brodcast(events.MAINTENANCE_MODE_  directly in the if / else clauses. The newGenericSettingProgressEventHandler doesn't actually seem to do anything.

That pattern is used for builds / group builds because they share some common behaviour. It isn't necessary here since it's just an abstraction over a single method call.